# argument = destination or product

int32 handle
int32 argument 
